optaplanner-examples/src/main/java/org/optaplanner/examples/flightcrewscheduling/domain/FlightAssignmentOrderComparator.java

```diff
@@ -0,0 +1,16 @@
+package org.optaplanner.examples.flightcrewscheduling.domain;
+
+import java.util.Comparator;
+
+public class FlightAssignmentOrderComparator implements Comparator<FlightAssignment> {
```

Comment:
AbstractPersistable should not be Comparable. Entities will be Comparable on-demand.

Reply:
@ge0ffrey This is going to be a massive change.

With `AbstractPersistable` no longer being `Comparable`, all uses of `CompareToBuilder` in examples need to be reviewed. `CompareToBuilder` doesn't require its arguments to be `Comparable`, and therefore - when comparing instances of `AbstractPersistable`, the new code could potentially fail at runtime.

I propose we still do this, but instead do it in a different PR. To not mix it with this work. (I will take the task.)